lib/postmark/api_client.rb

```diff
@@ -221,6 +221,22 @@ def deliver_with_template(attributes = {})
       end
     end
 
+    def get_stats_totals(options = {})
+      format_response(http_client.get('stats/outbound', options))
+    end
+
+    def get_stats_counts(stat, options = {})
+      url = "stats/outbound/#{stat}"
+
+      url += "/#{options.delete(:type)}" if options.has_key?(:type)
```

Comment:
I wouldn’t use `+=` here, as it creates a copy of joined strings. Since `url` is isolated to this method, it’s ok to use the mutating operator `<<`:

``` ruby
url << "/#{options.delete(:type)}" if options.has_key?(:type)
```

Also don’t use `delete` here as it mutates the arguments hash. Imagine the following scenario:

``` ruby
options = {type: 'user_agent'}
client.get_stats_counts(options) == client.get_stats_counts(options)
# => false (I think)
```